Realm/Realm/Linq/RealmResults.cs

```diff
@@ -79,14 +79,14 @@ internal override CollectionHandleBase GetOrCreateHandle()
         public override int IndexOf(T value)
         {
             Argument.NotNull(value, nameof(value));
-            
-            //TODO What to do here...?
-            //if (_argumentType != RealmValueType.Object) 
-            //{
-            //    throw new NotSupportedException("IndexOf on non-object results is not supported.");
-            //}
 
             var obj = value as RealmObjectBase;
+
+            if (obj == null)
```

Comment:
I think that previously we didn't have results that contain non-objects. Is it impossible to support `IndexOf` for non-object values similar to what we do for lists?